include/aspect/particle/property/interface.h

```diff
@@ -42,6 +42,36 @@ namespace aspect
     {
       using namespace dealii::Particles;
 
+      /**
+       * A data structure with all inputs for the
+       * Particle::Property::update_particle_properties() method.
+       */
+      template <int dim>
+      struct ParticleUpdateInputs
+      {
+        public:
+          /**
+           * The solution vector at each particle position. This vector is
+           * only filled if the update function requires the solution values.
+           */
+          small_vector<small_vector<double>> solution;
+
+          /**
+           * The solution gradients at each particle position.
+           * This vector is only filled if the update function requires the
+           * gradients of the solution values.
+           */
+          small_vector<small_vector<Tensor<1,dim>>> gradients;
```

Comment:
Are you sure that two nested small_vectors are a good choice? With the default arguments you are allocating 100*100 Tensors here. 
I think you can safely use an std on the outside or not?